profile-widget/profilewidget2.cpp

```diff
@@ -1822,12 +1822,20 @@ int ProfileWidget2::fixHandlerIndex(DiveHandler *activeHandler)
 	if (index > 0 && index < handles.count() - 1) {
 		DiveHandler *before = handles[index - 1];
 		if (before->pos().x() > activeHandler->pos().x()) {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 13, 0)
+			handles.swapItemsAt(index, index - 1);
```

Comment:
What about
```
std::swap(handles[index], handles[index - 1]);
```
? That should work for all Qt versions...

Reply:
won't that mess with Qt's weird ref counting semantics?

Reply:
OMG - I forgot about that particular insanity. Here, it shouldn't matter because when getting the first non-const reference,  the list goes into "detached" state.

Reply:
TBH, I'd be tempted to keep the ugliness that I have here as that doesn't risk mixing std::... and Qt memory management assumptions. Just doesn't seem worth it

Reply:
std::swap(&a,&b) literally just does
```
_tmp = a;
a = b;
b = _tmp;
```
If _that_ fails, we have more serious problems!

Here is the relevant snippet from qlist.h:
```
inline T &QList<T>::operator[](int i)
{ Q_ASSERT_X(i >= 0 && i < p.size(), "QList<T>::operator[]", "index out of range");
detach(); return reinterpret_cast<Node *>(p.at(i))->t(); }
```
With the first access, the list is copied (if it was shared) and therefore the second access goes to the same list.

This is safe. Do you have any particular reason to suspect that it isn't?

Reply:
no, just my regular Qt paranoia :-)

Reply:
ok, implemented that and pushed otherwise unmodified branch

Should I merge this once the CI passes?

Reply:
Go ahead...